### ISSUE TITLE
Is it possible to put .all-contributorsrc in subfolders?

### ISSUE THREAD

**Issue by laike9m** [all-contributors/all-contributors#464]
**Is your feature request related to a problem? Please describe.**
Right now it seems the `.all-contributorsrc` file can only be placed in the root folder. I would like to put it in the `.github` folder, is it possible?

**Describe the solution you'd like**
Not sure, but I feel we can relax the restriction by including `.github` as a valid location.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

**Comment by Berkmann18** [comment]
You make a good point, yeah, that feature would be useful. PR welcome.

**Comment by laike9m** [comment]
Is it possible to provide some guide, like, the code that reads this file. It would be helpful for whoever wants to work on this feature, it could be me or someone else.

**Comment by Berkmann18** [comment]
Sure, you'll need to look at https://github.com/all-contributors/all-contributors-cli and https://github.com/all-contributors/all-contributors-cli/blob/master/src/cli.js would be a good start.

**Comment by laike9m** [comment]
Thanks. I plan to work on bot -> cli -> documentation, because I use bot the most.

For implementation details, I'm thinking of the most straightforward way by adding a try-catch around the `getFile` call

https://github.com/all-contributors/all-contributors-bot/blob/97c0eea38164ce9e7a705e120af85124dc0a5c14/src/tasks/processIssueComment/OptionsConfig/index.js#L36-L39

So it would become something like:

```javascript
let content: rawOptionsFileContent;
let sha;
try {
    ({content, sha} = await this.repository.getFile(ALL_CONTRIBUTORS_RC));
} catch (error) {
    if (error instanceof ResourceNotFoundError) {
        ({content, sha} = await this.repository.getFile(".github/" + ALL_CONTRIBUTORS_RC));
    }
}
```

This way, no other code is affected, most of the time there's no overhead. And if users put the config file under `.github/`, at most there's one extra API call.

What do you think?

**Comment by Berkmann18** [comment]
> Thanks. I plan to work on bot -> cli -> documentation, because I use bot the most.

It may be wiser to try doing that for the CLI first. Why? Because the bot relies on it (to an extent).

Your implementation idea looks good to me.

**Comment by laike9m** [comment]
> It may be wiser to try doing that for the CLI first.

Sure if that's the case. May I ask how the bot is relying on the CLI?

**Comment by Berkmann18** [comment]
`./src/tasks/processIssueComment/OptionsConfig/index.js` and `./src/tasks/processIssueComment/ContentFiles/index.js` depends on it.

**Comment by GaryNLOL** [comment]
Hello. I wanted to know if a solution was finally found for this. I'm using All Contributors in my project but I don't want to have `.all-contributorsrc` in my root folder.

**Comment by laike9m** [comment]
@GaryNLOL No, I have not worked on it. Feel free to take over if you want to.
